### PR TITLE
added cassandra benchmarking models post

### PR DESCRIPTION
**Pull request by calonso**

### CODE REVIEW

File: posts/cassandra_models_benchmarking.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+During the last couple of weeks we've been very focused on deeply understanding Cassandra's
+write and read paths to build our models the best possible way.
+
+After so much investigation I can summarize which are the steps we follow to assess our
+model's performance.
+
+### 1. Have good practices and anti-patterns always in mind.
+
+Cassandra is known to be a very resilient and highly performant platform, but it will
```
Comment (gavinheavyside) [inline posts/cassandra_models_benchmarking.md]: , but only so long as you follow the rules and work *with* the underlying data model.

File: posts/cassandra_models_benchmarking.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+### 2. Design your data model and alternatives
+
+Following all the ideas and principles learnt in the previous step, design your data model and
+try to think of alternatives. These alternatives usually come as minor adjustments that can
+be applied to the model but just by following the best practices you can't decide whether one
+or the other is a better choice.
+
+Here I'll provide two examples we've recently had:
+
+1. Having a bucketed time series with a maximum of 86.400 rows per partition, how is it better to read an entire partition?
```
Comment (gavinheavyside) [inline posts/cassandra_models_benchmarking.md]: UK and US English usually uses commas for the thousands separator, so `86,400`

File: posts/cassandra_models_benchmarking.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+It is important that the access patterns are exactly the same way they'll be in
+production, otherwise the result of the benchmark is completely useless.
+
+This script should be adapted and ran for every single alternative.
+
+Here you can see the scripts used for both alternatives proposed for the example No. 2 described above:
+
+* [Benchmark the model with the list](https://gist.github.com/calonso/6dba841659000fdd7959)
+* [Benchmark the model with columns](https://gist.github.com/calonso/f14a0d9673110d6f9578)
```
Comment (gavinheavyside) [inline posts/cassandra_models_benchmarking.md]: Could you obfuscate the code in these examples please? Replace anything which might refer to internals, e.g. `rc` and `sub_id`

**Comment by gavinheavyside** [comment]
A few comments, but other than that looks good to me.

File: posts/cassandra_models_benchmarking.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+1. Having a bucketed time series with a maximum of 86.400 rows per partition, how is it better to read an entire partition?
+    a) By reading the whole partition at once
+    b) By reading the whole partition in chunks (2 halves, 4 quarters, ...)
+2. Having a model that contains the information of a discretised distribution on each record, how is it better to save the bins?
+    a) By using a *List* element that will contain the 100 bins
+    b) By having 100 columns, one for each bin
+
+The resulting models will meet all the good practices and avoid all the anti-patterns regardless
+of the final decission so, how do you decide which way to go?
```
Comment (mhefferan) [inline posts/cassandra_models_benchmarking.md]: decision

File: posts/cassandra_models_benchmarking.md (hunk 3 above)
Comment (gavinheavyside) [inline posts/cassandra_models_benchmarking.md]: I think I mean "sanitise" rather than "obfuscate".

**Comment by calonso** [comment]
Ready to review again :)

**Comment by mhefferan** [comment]
You still need to sanitize those Gists, but other than that looks good

**Comment by calonso** [comment]
more? :(

**Comment by mhefferan** [comment]
still some references to road_category in the second one?

On 15 May 2015 at 11:16, Carlos Alonso <notifications@github.com> wrote:

> more? :(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mydrive/mydrive.github.io/pull/6#issuecomment-102360963>
> .
>



-- 
_____________________

Michael Hefferan - Senior Software Engineer, MyDrive Solutions

-- 
www.mydrivesolutions.com

This email and any attachments is private and confidential. If you have 
received this message in error please remove it from your systems and 
notify the author.
MyDrive Solutions Limited is registered in England and Wales, No 07330334.
Registered address: Unit 12.G.1 The Leathermarket, Weston Street, London, 
SE1 3ER, UK.
VAT No: GB997011108

**Comment by calonso** [comment]
I think now all are gone

**Comment by mhefferan** [comment]
:shipit:

On 15 May 2015 at 11:28, Carlos Alonso <notifications@github.com> wrote:

> I think now all are gone
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mydrive/mydrive.github.io/pull/6#issuecomment-102362620>
> .
>



-- 
_____________________

Michael Hefferan - Senior Software Engineer, MyDrive Solutions

-- 
www.mydrivesolutions.com

This email and any attachments is private and confidential. If you have 
received this message in error please remove it from your systems and 
notify the author.
MyDrive Solutions Limited is registered in England and Wales, No 07330334.
Registered address: Unit 12.G.1 The Leathermarket, Weston Street, London, 
SE1 3ER, UK.
VAT No: GB997011108

**Pull request merged**